enabler/src/de/schildbach/pte/VrsProvider.java

```diff
@@ -0,0 +1,861 @@
+		STYLES.put("vrs|B611", new Style(Style.parseColor("#0099db"), Style.WHITE));
+		STYLES.put("vrs|B612", new Style(Style.parseColor("#ce9d53"), Style.WHITE));
+		STYLES.put("vrs|B613", new Style(Style.parseColor("#7b3600"), Style.WHITE));
+		STYLES.put("vrs|B614", new Style(Style.parseColor("#806839"), Style.WHITE));
+		STYLES.put("vrs|B615", new Style(Style.parseColor("#532700"), Style.WHITE));
+		STYLES.put("vrs|B630", new Style(Style.parseColor("#c41950"), Style.WHITE));
+		STYLES.put("vrs|B631", new Style(Style.parseColor("#9b1c44"), Style.WHITE));
+		STYLES.put("vrs|B633", new Style(Style.parseColor("#88cdc7"), Style.WHITE));
+		STYLES.put("vrs|B635", new Style(Style.parseColor("#cec800"), Style.WHITE));
+		STYLES.put("vrs|B636", new Style(Style.parseColor("#af0223"), Style.WHITE));
+		STYLES.put("vrs|B637", new Style(Style.parseColor("#e3572a"), Style.WHITE));
+		STYLES.put("vrs|B638", new Style(Style.parseColor("#af5836"), Style.WHITE));
+		STYLES.put("vrs|B640", new Style(Style.parseColor("#004f81"), Style.WHITE));
+		STYLES.put("vrs|BT650", new Style(Style.parseColor("#54baa2"), Style.WHITE));
+		STYLES.put("vrs|BT651", new Style(Style.parseColor("#005738"), Style.WHITE));
+		STYLES.put("vrs|BT680", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B800", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B812", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B843", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B845", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B852", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B855", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B856", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+		STYLES.put("vrs|B857", new Style(Style.parseColor("#4e6578"), Style.WHITE));
+
+		STYLES.put("vrs|S", new Style(Style.parseColor("#f18e00"), Style.WHITE));
+		STYLES.put("vrs|R", new Style(Style.parseColor("#009d81"), Style.WHITE));
+	}
+
+	public VrsProvider() {
+		super(NetworkId.VRS);
+		setStyles(STYLES);
+	}
+	
+	@Override
+	protected boolean hasCapability(Capability capability) {
+		switch (capability) {
+		case DEPARTURES:
+			return true;
+		case NEARBY_LOCATIONS:
+			return true;
+		case SUGGEST_LOCATIONS:
+			return true;
+		case TRIPS:
+			return true;
+		default:
+			return false;
+		}
+	}
+
+	// only stations supported
+	public NearbyLocationsResult queryNearbyLocations(EnumSet<LocationType> types /* only STATION supported */, Location location, int maxDistance /* filtered */, int maxLocations) throws IOException {
+		// g=p means group by product
+		final StringBuilder parameters = new StringBuilder();
+		parameters.append("?eID=tx_vrsinfo_ass2_timetable");
+		if (location.hasLocation()) {
+			parameters.append("&r=").append(location.lat / 1E6).append(",").append(location.lon / 1E6);
```

Comment:
I think it's safer to use "String.format(Locale.ENGLISH, "%.6f,%.6f", location.lon / 1E6, location.lat / 1E6))" here.